cmd/oras/login.go

```diff
@@ -131,26 +137,15 @@ func runLogin(opts loginOptions) (err error) {
 	return nil
 }
 
-func readLine(prompt string, silent bool) (string, error) {
-	fmt.Print(prompt)
-	if silent {
-		fd := os.Stdin.Fd()
-		state, err := term.SaveState(fd)
+func readLine(reader *bufio.Reader) (content []byte, err error) {
```

Comment:
To update the `term` package, we should touch the `readline` function only and should not touch the `runLogin`.

Reply:
>  we should touch the `readline` function only and should not touch the `runLogin`.

I have updated a version according to this suggestion. It's tricky that the newly added E2E specs are expected to fail since the bufio `reader` is generated within the function call, which consumes all mocked stdin during scanning for username and leaves password/token empty.

Can we at least make `reader` as parameter of  `readLine` so it's possible to do E2E tests via mocked stdin?

Reply:
fixed with self-implemented scanning